addons/event/models/event_registration.py

```diff
@@ -340,7 +341,7 @@ def _message_post_after_hook(self, message, msg_vals):
     def get_date_range_str(self):
         self.ensure_one()
         today = fields.Datetime.now()
-        event_date = self.event_begin_date
+        event_date = self.event_begin_date.astimezone(pytz.timezone(self.event_id.date_tz))
         diff = (event_date.date() - today.date())
```

Comment:
Hello there :wave:
Hum 2 points for this:
- The datetime in the database are stored in UTC and not in the timezone of the event. So here we should do: `pytz.utc.localize(self.event_begin_date).astimezone(pytz.timezone(self.event_id.date_tz))`. Because otherwise we just add the timezone without updating the datetime no ?
- Shouldn't we also do the same for `today` and apply the correct timezone. Because below we try to get a delta of two dates on possible different timezones which seems weird

Cheers!

Reply:
oof good catch
Oh I guess maybe it works because it's going the wrong way on the wrong date?

Time difference should handle different timezones fine so probably just UTC on event_date and the timezone on "today" instead and forget "astimezone" should work

Reply:
Hi folks, appreciate the additional eyes! I agree that comparing the same timezones would be the best option, which would look like:

`today = pytz.utc.localize(fields.Datetime.now()).astimezone(pytz.timezone(self.event_id.date_tz))`
`event_date = pytz.utc.localize(self.event_begin_date).astimezone(pytz.timezone(self.event_id.date_tz))`

With some whiteboarding, I'm not sure when a test case would fail (when UTC changes date first) given the mocked `datetime(2020, 1, 31, 10, 0, 0)` Today. I think an event with `datetime(2020, 2, 5, 12, 0, 0)` would check the opposite case though. Thoughts?

Reply:
Yes it's indeed what I had in mind.
Also an example for the test case would be:

today = 2023-06-28 06:00 (UTC)
event_start = 2023-06-28 16:00 (UTC)

If the timezone of the event is PST then we will have

today = 2023-06-27 23:00 (PST)
event_start = 2023-06-28 09:00 (PST)

which before (in UTC) would return "today" and after (in PST) would return "tomorrow" as the date are no longer the same.

Ideally it would be better to test both extreme with different timezone to be consistent (like PST above and another like AEST perhaps)